lib/profile/commands/apply.rb

```diff
@@ -173,6 +177,28 @@ def check_nodes_exist(names=[])
           raise out
         end
       end
+      
+      def expand_brackets(str)
+        labels = []
+        contents = str[/\[.*\]/] 
+        if contents.nil?
+          labels = [str]
+        else
+          left = str[/[^\[]*/]
+          right = str[/].*/][1..-1]
+          if contents.match(/^\[[0-9]+-[0-9]+\]$/)
+            nums = contents[1..-2].split("-")
+            (nums.last.to_i - nums.first.to_i + 1).times do |index|
```

Comment:
Giving the method a range where the start is greater than the end (e.g. `node[5-1]`) results in an empty array being returned to the main `apply` logic. Do you think we ought to leave it like that or inform the user that the upper bound ought to be greater than the lower bound?

Reply:
That's worth throwing an error for I think, I can't envision this scenario ever intentionally occurring, it'd most likely be a typo. If a user DID type it in intentionally it certainly isn't good behaviour to not apply to any nodes.